console/src/telemetry/index.ts

```diff
@@ -0,0 +1,109 @@
+import endpoints from '../Endpoints';
+import globals from '../Globals';
+import { filterEventsBlockList, sanitiseUrl } from './filters';
+import { RUN_TIME_ERROR } from '../components/Main/Actions';
+import { REDUX_LOCATION_CHANGE_ACTION_TYPE } from '../constants';
+
+type ReduxAction = {
+  type: string;
+  payload: {
+    pathname: string;
+  };
+  data: any;
+};
+
+interface TelemetryGlobals {
+  serverVersion: string;
+  consoleMode: string;
+  cliUUID: string;
+  hasuraUUID: string;
+}
+
+const createClient = () => {
+  if (globals.enableTelemetry) {
+    try {
+      const client = new WebSocket(endpoints.telemetryServer);
+      client.onerror = e => {
+        console.error(`WebSocket Error for Events${e}`);
+      };
+      return client;
+    } catch (e) {
+      console.error('Unable to initialise telemetry client', e);
+      return null;
+    }
+  }
+  return null;
+};
+
+let client = createClient();
+if (client) {
+  const onClose = () => {
+    client = createClient();
+    if (client) {
+      client.onclose = onClose;
+    }
+  };
+  client.onclose = onClose;
+}
+
+const isTelemetryConnectionReady = () => {
```

Comment:
Can't it be a variable? There are no arguments anyway.

Reply:
No. The variable will be a constant and never changed. We want to get the connection ready state at the instance where this is being checked.